### DISCUSSION TITLE
teleport jwt opensearch

### DISCUSSION THREAD

**Discussion by Bah27** [gravitational/teleport#36125] [Q&A]
Hello, I would like to configure jwt authentication with teleport on a siem wazuh with opensearch. do you have a configuration example? I tried but I can't get it. Thanks in advance for your help.

**Comment by webvictim** [comment]
Could you share some more detail about what you've tried and any errors you're seeing?

**Reply by Bah27** [reply]
Good evening, thank you for your feedback. My configuration is as follows:
teleport configuration: 
apps:
  - name: wazuh-dashboard
    uri: https://dashboard.wazuh.svc.cluster.local:443
    insecure_skip_verify: true
    labels:
      app: wazuh-dashboard
    rewrite:
      jwt_claims: roles
      headers:
      - "Authorization: Bearer {{internal.jwt}}"
      
configuration wazuh opensearch:
jwt_auth_domain:
        description: "Authenticate via Json Web Token"
        http_enabled: true
        transport_enabled: true
        order: 0
        http_authenticator:
          type: jwt
          challenge: false
          config:
            signing_key: |-
              -----BEGIN PUBLIC KEY-----
              ----
              -----END PUBLIC KEY-----
            jwt_header: "Teleport-Jwt-Assertion"
            jwt_url_parameter: null
            jwt_clock_skew_tolerance_seconds: 30
            roles_key: "roles"
            subject_key: "sub"
        authentication_backend:
          type: noop

Error returned:
{
    "statusCode": 401,
    "error": "Unauthorized",
    "message": "Service Unavailable"
}

When I check the logs, I get the following output:
{"type":"response","@timestamp":"2024-01-21T20:38:22Z","tags":[],"pid":54,"method":"get","statusCode":401,"req":{"url":"/","method":"get","headers":{"host":"dashboard.wazuh.svc.cluster.local:443","user-agent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36 Edg/120.0.0.0","accept":"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7","accept-encoding":"gzip, deflate, br","accept-language":"fr,fr-FR;q=0.9,en;q=0.8,en-GB;q=0.7,en-US;q=0.6","sec-ch-ua":"\"Not_A Brand\";v=\"8\", \"Chromium\";v=\"120\", \"Microsoft Edge\";v=\"120\"","sec-ch-ua-mobile":"?0","sec-ch-ua-platform":"\"Windows\"","sec-fetch-dest":"document","sec-fetch-mode":"navigate","sec-fetch-site":"same-site","teleport-jwt-assertion":"-------------------------","upgrade-insecure-requests":"1","x-forwarded-for":"------","x-forwarded-host":"-------","x-forwarded-port":"443","x-forwarded-proto":"https","x-forwarded-scheme":"https","x-forwarded-server":"teleport-agent-0","x-forwarded-ssl":"on","x-real-ip":"----------","x-request-id":"f011ffa419cfcf10e0860d7153938724","x-scheme":"https"},"remoteAddress":"---------","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36 Edg/120.0.0.0"},"res":{"statusCode":401,"responseTime":2464,"contentLength":9},"message":"GET / 401 2464ms - 9.0B"}

I referred to the following opensearch documentation: [https://opensearch.org/docs/latest/security/authentication-backends/jwt/](https://opensearch.org/docs/latest/security/authentication-backends/jwt/) .

Thanks in advance
